packages/next-swc/crates/next-core/src/manifest.rs

```diff
@@ -77,10 +80,17 @@ impl DevManifestContentSourceVc {
         let mut routes = routes
             .into_iter()
             .flatten()
-            .map(|s| s.to_string())
+            .map(|s| {
+                if !s.starts_with('/') {
+                    format!("/{}", s)
+                } else {
+                    s.to_string()
+                }
```

Comment:
That seem to be hiding an inconsistency in `get_pathname()`.
* Add a comment to `get_pathname()` about the intended behavior (seems like with / should be always used)
* Make all `get_pathname` implementations run the correct format
* Potentially assert here that it has a leading /